lib/themes/dosomething/paraneue_dosomething/js/signup/Signup.js

```diff
@@ -22,53 +31,150 @@ const Signup = {
+   * @param  {bool}   slide       Whether or not to add the takeover with a slide animation.
+   */
+  addTakeover: function($container, data, slide = false) {
+    const markup = template(takeoverTpl);
+    $container.html(markup(data));
     this.takeoverContent = $('.takeover');
-    this.takeoverContent.hide();
-    // this.takeoverContent.slideDown("slow", function() {});
-    this.takeoverContent.delay(1000).slideDown("slow", function() {});
 
-   const $signupButton = $('.signup-button');
+    if (slide) {
+      this.takeoverContent.hide();
+      this.takeoverContent.delay(500).slideDown("slow", function() {});
+    }
+  },
+
+  /**
+   * Binds click handlers to signup buttons that posts signs ups to phoenix.
+   *
+   * @param  {jQuery}   $button   Signup buttons
+   */
+  handleSignups: function($button) {
+    const _this = this;
+
+    // Create custom deferred promise object.
+    let ajaxPromise = $.Deferred().resolve().promise();
+
+    // Bind click handler
+    $button.click(function(e) {
+      e.preventDefault();
 
-   $signupButton.click(function() {
       const $this = $(this);
-      const campaignId = $this.data('campaign-id');
-
-      $.ajax({
-        method: 'POST',
-        url: '/api/v1/campaigns/' + campaignId +'/signup', //id ? `/api/v1/kudos/${id}` : '/api/v1/kudos',
-        dataType: 'json',
-        headers: {
-          'Accepts': 'application/json',
-          'Content-Type': 'application/json',
-          'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
-        },
-        data: JSON.stringify({
-          'uid': $('meta[name="drupal-user-id"]').attr('content'),
-          'source': 'multi-signup-module',
-        }),
-        beforeSend: function() {
-          console.log("beforeSend");
-          const $buttonContainer = $this.closest('.takeover__campaign-item');
-          $buttonContainer.html("<div class='spinner'></div>");
-        },
-        success: function(data) {
-          console.log('success');
-          console.log(data);
-
-          const $buttonContainer = $('.spinner').closest('.takeover__campaign-item');
-          console.log($buttonContainer);
-          $buttonContainer.html('<p>Signed Up!</p>');
+      const $buttonContainer = $this.closest('.-signup');
+
+      // Make the ajax call, store the promise.
+      const promise = _this.postSignup($this);
+
+      // Fire done call backs in the order in which they were recieved.
+      ajaxPromise = ajaxPromise.then(function() {
+        return promise;
+      }).done(function(data) {
+        const signupId = data[0];
+
+        if (signupId) {
+          $this.val(_this.checkmark);
+          $this.css('background-color', _this.green);
+          $buttonContainer.html($this);
+        // @TODO - How should we handle campaigns folks are already signed up for.
+        // Do we even need to handle this case, ideally we would only show campaigns they
+        // are not signed up for.
```

Comment:
@sbsmith86 `dosomething_campaign_get_recommended_campaign_nids` function handles that [here](https://github.com/DoSomething/phoenix/blob/bb4fa06179e98a3b76cc14dd5a4791edfc8cf3e1/lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module#L1028)!